src/umleditor/mvc_model/custom_exceptions.py

```diff
@@ -76,6 +76,32 @@ class RelationDoesNotExistError(Error):
         def __init__(self, source, destination):
             super().__init__(f"Relation between '{source} -> {destination}' does not exist.")
 
+    class InvalidRelationTypeError(Error):
+        """
+        Exception raised when the relation being added has no types.
```

Comment:
This will be caught by the parser and reported as too few arguments before it ever reaches the relation methods

Reply:
To be clear, only none will be checked by the parser. For a relation to have no type, it needs to not have a param passed in that spot. When the parser tries to call add relation with two params instead of three, an instance of TypeError will be raised, caught, and replaced by an instance of CustomExceptions.InvalidArgCountError.

Reply:
^ as something actionable that can be changed, this error can be removed - it is unreachable.